streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -546,6 +555,91 @@ public void foreach(final ForeachAction<? super K, ? super V> action,
         );
     }
 
+    @Override
+    public KStream<K, V> repartition() {
+        return doRepartition(Repartitioned.as(null), null);
+    }
+
+    @Override
+    public KStream<K, V> repartition(final Repartitioned<K, V> repartitioned) {
+        Objects.requireNonNull(repartitioned, "repartitioned parameter can't be null");
```

Comment:
Thanks for the suggestion. We can do null check in `doRepartition` for `repartitioned` parameter. null check for `selector` parameter should be on upper level since in some cases we do pass selector as `null`. For example:
```java
    @Override
    public KStream<K, V> repartition() {
        return doRepartition(Repartitioned.as(null), null);
    }
```

Will update the error msg as well.